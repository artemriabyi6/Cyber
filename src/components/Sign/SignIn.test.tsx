import React from 'react';
import { renderToString } from 'react-dom/server';
import SignInForm from './SignIn';



describe('SignInForm Component', () => {
  test('рендериться без помилок', () => {
    const htmlString = renderToString(<SignInForm />);
    expect(htmlString).toBeTruthy();
  });

  test('містить поля для входу', () => {
    const htmlString = renderToString(<SignInForm />);
    
    // Перевіряємо наявність ключових елементів (з урахуванням HTML кодування)
    expect(htmlString).toContain('Email');
    expect(htmlString).toContain('Пароль');
    expect(htmlString).toContain('Запам&#x27;ятати мене');
    expect(htmlString).toContain('Забули пароль?');
  });

  test('містить кнопку входу', () => {
    const htmlString = renderToString(<SignInForm />);
    expect(htmlString).toContain('Увійти');
  });

  test('містить посилання на реєстрацію', () => {
    const htmlString = renderToString(<SignInForm />);
    expect(htmlString).toContain('Не маєте акаунта?');
    expect(htmlString).toContain('href="/signup"');
  });

  test('відповідає snapshot', () => {
    const htmlString = renderToString(<SignInForm />);
    expect(htmlString).toMatchSnapshot();
  });
});