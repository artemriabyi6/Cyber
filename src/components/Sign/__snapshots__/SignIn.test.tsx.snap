// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SignInForm Component відповідає snapshot 1`] = `"<div class="min-h-screen bg-gradient-to-br from-blue-50 to-indigo-100 flex items-center justify-center px-4 py-8"><div class="max-w-md w-full space-y-8"><div class="bg-white rounded-2xl shadow-xl p-8"><div class="text-center mb-8"><h2 class="text-3xl font-bold text-gray-900">Ласкаво просимо</h2><p class="mt-2 text-gray-600">Увійдіть у свій акаунт</p></div><form class="space-y-6"><div><label for="email" class="block text-sm font-medium text-gray-700 mb-1">Email</label><input id="email" type="email" class="w-full px-4 py-3 border rounded-lg focus:ring-2 focus:ring-blue-500 focus:border-transparent transition-all border-gray-300" placeholder="your@email.com" name="email" value=""/></div><div><label for="password" class="block text-sm font-medium text-gray-700 mb-1">Пароль</label><input id="password" type="password" class="w-full px-4 py-3 border rounded-lg focus:ring-2 focus:ring-blue-500 focus:border-transparent transition-all border-gray-300" placeholder="Ваш пароль" name="password" value=""/></div><div class="flex items-center justify-between"><div class="flex items-center"><input id="rememberMe" type="checkbox" class="w-4 h-4 text-blue-600 border-gray-300 rounded focus:ring-blue-500" name="rememberMe"/><label for="rememberMe" class="ml-2 block text-sm text-gray-700">Запам&#x27;ятати мене</label></div><div class="text-sm"><a class="text-blue-600 hover:text-blue-500 font-medium" href="/forgot-password">Забули пароль?</a></div></div><button type="submit" class="w-full bg-gradient-to-r from-blue-600 to-indigo-600 text-white py-3 px-4 rounded-lg font-semibold hover:from-blue-700 hover:to-indigo-700 transform hover:-translate-y-0.5 transition-all duration-200 shadow-lg hover:shadow-xl">Увійти</button><div class="text-center"><p class="text-gray-600">Не маєте акаунта?<!-- --> <a class="text-blue-600 hover:text-blue-500 font-medium" href="/signup">Зареєструватись</a></p></div></form><div class="mt-6"><div class="relative"><div class="absolute inset-0 flex items-center"><div class="w-full border-t border-gray-300"></div></div><div class="relative flex justify-center text-sm"><span class="px-2 bg-white text-gray-500">Або продовжити з</span></div></div><div class="mt-6 grid grid-cols-2 gap-3"><button type="button" class="w-full inline-flex justify-center py-2 px-4 border border-gray-300 rounded-md shadow-sm bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"><span>Google</span></button><button type="button" class="w-full inline-flex justify-center py-2 px-4 border border-gray-300 rounded-md shadow-sm bg-white text-sm font-medium text-gray-500 hover:bg-gray-50"><span>GitHub</span></button></div></div></div></div></div>"`;
